### ISSUE TITLE
Replace Panel with another API to feat InstantSearch.css requirements 

### ISSUE THREAD

**Issue by mthuret** [algolia/react-instantsearch#503]
We want to have the same markup for every InstantSearch. To hide a facets title if no requirements we currently have two different strategy between IS.js and RIS. 

We want to converge to a `header/body/footer` model for every components. 

Ex: 

```html
<div class="ais-Menu">
  <div class="ais-Menu-header ais-header">
    Header (optional)
  </div>
  <div class="ais-Menu-body ais-body">
    <ul class="ais-Menu-list">
      <li class="ais-Menu-item ais-Menu-item--selected">
        <a class="ais-Menu-link" href="some-link">
          Appliances
          <span class="ais-Menu-count">4,306</span>
        </a>
      </li>
      <li class="ais-Menu-item">
        <a class="ais-Menu-link" href="some-link">
          Audio
          <span class="ais-Menu-count">1,570</span>
        </a>
      </li>
    </ul>
  </div>
  <div class="ais-Menu-footer ais-footer">
    Footer (optional)
  </div>
</div>
```

## API proposal:

* deprecate `<Panel/>`
* add two props `header` and `footer` that can take a React Component.

**Comment by mthuret** [comment]
Sure! a function that has the props and expect a React component

**Comment by samouss** [comment]
Closed by #833, but we keep the behaviour of RIS. We add a Panel in IS.css for being capable to compose widgets together.
